tests/sqlparser_postgres.rs

```diff
@@ -2477,6 +2477,257 @@ fn parse_create_anonymous_index() {
     }
 }
 
+#[test]
+fn parse_create_users_name_trgm_index() {
+    let sql = "CREATE INDEX users_name_trgm_idx ON users USING GIN (concat_users_name(first_name, last_name) gin_trgm_ops)";
```

Comment:
If I understood the syntax, its possible to have multiple indexed columns, could we have a test covering that if so? Also it would be ideal if we included test coverage for the various index types and operator classes introduced (though for the latter I think if we switch to opaque identifiers that would not be necessary).

(Also note that it would probably be easier to add further tests without asserting the full AST, only relying on verified_stmt roundtrip)